src/yb/tools/ts-cli.cc

```diff
@@ -194,15 +208,22 @@ TsAdminClient::~TsAdminClient() {
 Status TsAdminClient::Init() {
   CHECK(!initted_);
 
-  HostPort host_port;
-  RETURN_NOT_OK(host_port.ParseString(addr_, tserver::TabletServer::kDefaultPort));
-  messenger_ = VERIFY_RESULT(MessengerBuilder("ts-cli").Build());
-
-  rpc::ProxyCache proxy_cache(messenger_.get());
-
-  generic_proxy_.reset(new server::GenericServiceProxy(&proxy_cache, host_port));
-  ts_proxy_.reset(new TabletServerServiceProxy(&proxy_cache, host_port));
-  ts_admin_proxy_.reset(new TabletServerAdminServiceProxy(&proxy_cache, host_port));
+  if(!certs_dir_.empty()) {
+    HostPort host_port;
+    RETURN_NOT_OK(host_port.ParseString(addr_, tserver::TabletServer::kDefaultPort));
+    MessengerBuilder messenger_builder("ts-cli");
+    FLAGS_use_client_to_server_encryption = true;
+    FLAGS_certs_dir = certs_dir_;
```

Comment:
Why flags is changed here? `CreateSecureContext` can be used instead of `SetupSecureContext` if for some reason actual value of flags can't be used to enable encryption.

Reply:
CreateSecureContext is not used as we need to applySecureContext to message builder after creation which is effectively done by SetupSecureContext